ee/tabby-webserver/src/schema/mod.rs

```diff
@@ -598,6 +604,49 @@ impl Mutation {
         ctx.locator.license().reset_license().await?;
         Ok(true)
     }
+
+    async fn daily_stats_in_past_year(
+        ctx: &Context,
+        users: Option<Vec<ID>>,
+    ) -> Result<Vec<CompletionStats>> {
+        let users = users.unwrap_or_default();
+        check_analytic_access(ctx, &users).await?;
+        ctx.locator.analytic().daily_stats_in_past_year(users).await
+    }
+
+    async fn daily_stats(
+        ctx: &Context,
+        start: DateTime<Utc>,
+        end: DateTime<Utc>,
+        users: Option<Vec<ID>>,
+        languages: Option<Vec<analytic::Language>>,
+    ) -> Result<Vec<CompletionStats>> {
+        let users = users.unwrap_or_default();
+        check_analytic_access(ctx, &users).await?;
+        ctx.locator
+            .analytic()
+            .daily_stats(start, end, users, languages.unwrap_or_default())
+            .await
+    }
+}
+
+async fn check_analytic_access(ctx: &Context, users: &[ID]) -> Result<(), CoreError> {
```

Comment:
Add test to cover this function, I don't think it is going to work as intended

Reply:
ack. prefer not to add mock around `Context` for now.